vpr/src/place/timing_place_lookup.cpp

```diff
@@ -220,22 +220,24 @@ std::vector<int> get_best_classes(enum e_pin_type pintype, t_physical_tile_type_
      */
 
     std::vector<int> best_classes;
-
+    int best_class_index = 0;
     //Collect all classes of matching type which do not have all their pins ignored
-    for (int i = 0; i < type->num_class; i++) {
-        if (type->class_inf[i].type == pintype) {
+    for (auto class_inf : type->class_inf) {
```

Comment:
Looks like this will be copying the class_inf, probably want to use `const auto&` to keep the same reference only behavior of the previous code.

Reply:
Done